camel-core/src/main/java/org/apache/camel/component/dataset/DataSetEndpoint.java

```diff
@@ -59,6 +60,8 @@
     private long preloadSize;
     @UriParam(label = "consumer", defaultValue = "1000")
     private long initialDelay = 1000;
+    @UriParam(label = "consumer,producer", defaultValue = "null")
+    private Boolean disableDataSetIndex;
```

Comment:
can we flip this, as inverted options is confusing. So by default it can be true instead. dataSetIndex=true as default. And label="consumer,producer" is implied so you do not need that. Also defaultValue = null is also implied, eg no default value.